src/darker/tests/conftest.py

```diff
@@ -26,20 +27,25 @@ def with_isort():
 
 
 class GitRepoFixture:
-    def __init__(self, root: Path):
+    def __init__(self, root: Path, env: Dict[str, str]):
         self.root = root
+        self.env = env
 
     @classmethod
     def create_repository(cls, root: Path) -> "GitRepoFixture":
         """Fixture method for creating a Git repository in the given directory"""
-        check_call(["git", "init"], cwd=root)
-        check_call(["git", "config", "user.email", "ci@example.com"], cwd=root)
-        check_call(["git", "config", "user.name", "CI system"], cwd=root)
-        return cls(root)
+        env = os.environ.copy()
+        # for testing, ignore ~/.gitconfig settings like templateDir and defaultBranch
+        env["HOME"] = str(root)
+        instance = cls(root, env)
+        instance._run("init")
```

Comment:
PyLint complains and fails our build unless we explicitly allow access to a protected method of another class instance:
```suggestion
        # pylint: disable=protected-access
        instance._run("init")
```

Reply:
I added this, too, but for what it's worth, didn't see a complaint when running the test suite locally or via `darker -L pylint`  with black version 21.5b0 on Python 3.7.3.

Reply:
Strange, I did see the complaint. Maybe we have different versions of Pylint.